File: ai-project-back-end/app/schemas/workspace.py
from __future__ import annotations

from typing import Literal

from pydantic import Field

from app.schemas.common import BaseSchema
from app.schemas.types import UnixTs


class WorkspaceAssetSummary(BaseSchema):
    requirementDocs: int = Field(ge=0)
    testcases: int = Field(ge=0)
    formalCases: int = Field(ge=0)
    testPoints: int = Field(ge=0)
    apiCollections: int = Field(ge=0)
    apiRequests: int = Field(ge=0)
    suites: int = Field(ge=0)


class WorkspaceAutomationSummary(BaseSchema):
    runs: int = Field(ge=0)
    executedCaseRuns: int = Field(ge=0)
    passRate: float = Field(ge=0, le=100)
    latestRunAt: UnixTs | None = None


class WorkspaceRiskSummary(BaseSchema):
    defects: int = Field(ge=0)
    p0Open: int = Field(ge=0)
    riskHints: int = Field(ge=0)


class WorkspaceCapabilitySummary(BaseSchema):
    role: Literal["admin", "owner", "editor", "viewer"]
    assets: bool
    ai: bool
    automation: bool
    settings: bool
    ops: bool


class WorkspaceSummaryData(BaseSchema):
    assets: WorkspaceAssetSummary
    automation: WorkspaceAutomationSummary
    risks: WorkspaceRiskSummary
    capabilities: WorkspaceCapabilitySummary

File: ai-project_front_end/src/lib/api/workspace.ts
import { authHeader, requestJson } from '@/lib/api/client'

export type WorkspaceSummary = {
  assets: {
    requirementDocs: number
    testcases: number
    formalCases: number
    testPoints: number
    apiCollections: number
    apiRequests: number
    suites: number
  }
  automation: {
    runs: number
    executedCaseRuns: number
    passRate: number
    latestRunAt: number | null
  }
  risks: {
    defects: number
    p0Open: number
    riskHints: number
  }
  capabilities: {
    role: 'admin' | 'owner' | 'editor' | 'viewer'
    assets: boolean
    ai: boolean
    automation: boolean
    settings: boolean
    ops: boolean
  }
}

export async function fetchWorkspaceSummary(projectId: string) {
  const pid = String(projectId || '').trim()
  if (!pid) throw new Error('projectId 不能为空')
  return requestJson<WorkspaceSummary>(`/api/projects/${encodeURIComponent(pid)}/workspace/summary`, {
    method: 'GET',
    headers: authHeader()
  })
}
